osu.Game/Graphics/Cursor/OsuCursorVisualiser.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System.Linq;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Cursor;
+using osu.Framework.Input;
+
+namespace osu.Game.Graphics.Cursor
+{
+    /// <summary>
+    /// Visualises different cursors depending on the currently-hovered <see cref="Drawable"/>s.
+    /// </summary>
+    public class OsuCursorVisualiser : Container, IProvideCursor

```

Comment:
we really need to come up with a consistent naming style for these things (there's also things like `TooltipContainer` which is a similar pattern, but different name).

Reply:
Yeeeeeah I named it this way because I don't want to make it seem like you should be using an "OsuCursorContainer" instead of a "CursorContainer". Suggestions welcome.